vm-virtio/src/pmem.rs

```diff
@@ -333,24 +329,29 @@ impl Pmem {
             avail_features |= 1u64 << VIRTIO_F_IOMMU_PLATFORM;
         }
 
-        Ok(Pmem {
+        let pmem_backend = VirtioBackend {
             kill_evt: None,
-            pause_evt: None,
-            disk: Some(disk),
             avail_features,
             acked_features: 0u64,
-            config,
+            config_space: Vec::new(),
             queue_evts: None,
             interrupt_cb: None,
+        };
+
+        Ok(Pmem {
+            pause_evt: None,
+            disk: Some(disk),
+            config,
```

Comment:
This is a problem here because we have an empty/ignored `config_space` in the `VirtioBackend` while the real `config` is stored into the `Pmem` structure. You should be able to reuse `config_space`, otherwise the `VirtioBackend` abstraction is not well defined.

Reply:
Fixed.